src/screens/ChannelListScreenV2/index.tsx

```diff
@@ -37,6 +42,81 @@ const ChannelListScreenV2 = () => {
     }
   };
 
+  const requestPermission = async () => {
+    const authStatus = await messaging().requestPermission();
+    const enabled =
+      authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
+      authStatus === messaging.AuthorizationStatus.PROVISIONAL;
+
+    if (enabled) {
+      const fcmToken = await messaging().getToken();
+      const payload = {
+        fcm_token: fcmToken
+      };
+      fcmTokenService(payload);
+    }
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestPermission` function is requesting the FCM token and sending it to the server. However, there's no error handling in case either of these operations fail. It would be a good idea to wrap these calls in a try-catch block to handle potential errors.

```diff
  const requestPermission = async () => {
+   try {
      const authStatus = await messaging().requestPermission();
      const enabled =
        authStatus === messaging.AuthorizationStatus.AUTHORIZED ||
        authStatus === messaging.AuthorizationStatus.PROVISIONAL;

      if (enabled) {
        const fcmToken = await messaging().getToken();
        const payload = {
          fcm_token: fcmToken
        };
        fcmTokenService(payload);
      }
+   } catch (error) {
+     console.error('Error while requesting permission or getting FCM token:', error);
+   }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->